scripts/autolabel.lean

```diff
@@ -0,0 +1,160 @@
+* The `dirs` field is the array of all "root paths" such that a modification in a file contained
+  in one of these paths should be labeled by `label`.
+* The `exclusions` field is the array of all "root paths" that are excluded, among the
+  ones that start with the ones in `dirs`.
+-/
+structure Label where
+  /-- The label name as it appears on github -/
+  label : String
+  /-- Array of paths which fall under this label. e.g. `"Mathlib" / "Algebra"` -/
+  dirs : Array System.FilePath
+  /-- Array of (sub)-paths which should be excluded -/
+  exclusions : Array System.FilePath := #[]
+  deriving BEq, Hashable
+
+/--
+Mathlib Labels and their corresponding folders. Add new labels and folders here!
+-/
+private def mathlibLabels : Std.HashSet Label := Std.HashSet.ofList [
+  { label := "t-algebra",
+    dirs := #[
+      "Mathlib" / "FieldTheory",
+      "Mathlib" /" RingTheory",
+      "Mathlib" / "GroupTheory",
+      "Mathlib" / "RepresentationTheory",
+      "Mathlib" / "LinearAlgebra" ]},
+  { label := "t-algebraic-geometry",
+    dirs := #[
+      "Mathlib" / "AlgebraicGeometry",
+      "Mathlib" / "Geometry.RingedSpace" ]},
+  { label := "t-analysis",
+    dirs := #[
+      "Mathlib" / "Analysis" ]},
+  { label := "t-category-theory",
+    dirs := #[
+      "Mathlib" / "CategoryTheory" ]},
+  { label := "t-combinatorics",
+    dirs := #[
+      "Mathlib" / "Combinatorics" ]},
+  { label := "t-computability",
+    dirs := #[
+      "Mathlib" / "Computability" ]},
+  { label := "t-condensed",
+    dirs := #[
+      "Mathlib" / "Condensed" ]},
+  { label := "t-data",
+    dirs := #[
+      "Mathlib" / "Data" ]},
+  { label := "t-differential-geometry",
+    dirs := #[
+      "Mathlib" / "DifferentialGeometry",
+      "Mathlib" / "Geometry" / "Manifold" ]},
+  { label := "t-dynamics",
+    dirs := #[
+      "Mathlib" / "Dynamics"]},
+  { label := "t-euclidean-geometry",
+    dirs := #[
+      "Mathlib" / "Geometry" / "Euclidean" ] },
+  { label := "t-linter",
+    dirs := #[
+      "Mathlib" / "Tactic" / "Linter" ]},
+  { label := "t-logic",
+    dirs := #[
+      "Mathlib" / "Logic",
+      "Mathlib" / "ModelTheory" ]},
+  { label := "t-measure-probability",
+    dirs := #[
+      "Mathlib" / "MeasureTheory",
+      "Mathlib" / "Probability",
+      "Mathlib" / "InformationTheory" ]},
+  { label := "t-meta",
+    dirs := #[
+      "Mathlib" / "Tactic" ],
+    exclusions := #[
+      "Mathlib" / "Tactic" / "Linter" ]},
+  { label := "t-number-theory",
+    dirs := #[
+      "Mathlib" / "NumberTheory"]},
+  { label := "t-order",
+    dirs := #[
+      "Mathlib" / "Order" ]},
+  { label := "t-set-theory",
+    dirs := #[
+      "Mathlib" / "SetTheory"]},
+  { label := "t-topology",
+    dirs := #[
+      "Mathlib" / "Topology",
+      "Mathlib" / "AlgebraicTopology" ]},
+  { label := "CI",
+    dirs := #[
+      ".github" / "workflows" ]}]
+
+def getMatchingLabels (files : Array String) : Array Label :=
```

Comment:
Would you mind adding a doc-string to this definition?  It shows that I've been playing with linters for a while, doesn't it?  :smile: